### ISSUE TITLE
Error: Request had insufficient authentication scopes

### ISSUE THREAD

**Issue by tsteur** [matomo-org/plugin-GoogleAnalyticsImporter#113]
When running `googleanalyticsimporter:import-reports --idsite=X`

Not sure if this error is somehow avoidable @diosmosis ? Eg when configuring an account can we check needed permissions are there?

Saw this error in the logs:

```
Error: {"message":"{\n \"error\": {\n \"code\": 403,\n \"message\": \"Request had insufficient authentication scopes.\",\n \"errors\": [\n {\n \"message\": \"Insufficient Permission\",\n \"domain\": \"global\",\n \"reason\": \"insufficientPermissions\"\n }\n ],\n \"status\": \"PERMISSION_DENIED\"\n }\n}\n","file":"\/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient\/src\/Google\/Http\/REST.php","line":118,"backtrace":" on \/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient\/src\/Google\/Http\/REST.php(118)\n#0 \/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient\/src\/Google\/Http\/REST.php(94): Google_Http_REST::decodeHttpResponse(Object(GuzzleHttp\\Psr7\\Response), Object(GuzzleHttp\\Psr7\\Request), 'Google_Service_...')\n#1 \/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient\/src\/Google\/Task\/Runner.php(176): Google_Http_REST::doExecute(Object(GuzzleHttp\\Client), Object(GuzzleHttp\\Psr7\\Request), 'Google_Service_...')\n#2 \/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient\/src\/Google\/Http\/REST.php(58): Google_Task_Runner->run()\n#3 \/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient\/src\/Google\/Client.php(862): Google_Http_REST::execute(Object(GuzzleHttp\\Client), Object(GuzzleHttp\\Psr7\\Request), 'Google_Service_...', Array, NULL)\n#4 \/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient\/src\/Google\/Service\/Resource.php(232): Google_Client->execute(Object(GuzzleHttp\\Psr7\\Request), 'Google_Service_...')\n#5 \/plugins\/GoogleAnalyticsImporter\/vendor\/google\/apiclient-services\/src\/Google\/Service\/Analytics\/Resource\/ManagementGoals.php(84): Google_Service_Resource->call('list', Array, 'Google_Service_...')\n#6 \/plugins\/GoogleAnalyticsImporter\/Importer.php(213): Google_Service_Analytics_Resource_ManagementGoals->listManagementGoals('YYYYYYY', 'UA-XXXXXXXX-X', 'XXXXXX')\n#7 \/plugins\/GoogleAnalyticsImporter\/Importer.php(194): Piwik\\Plugins\\GoogleAnalyticsImporter\\Importer->importGoals('2', 'YYYYYYY', 'UA-XXXXXXXX-X', 'XXXX')\n#8 \/plugins\/GoogleAnalyticsImporter\/Commands\/ImportReports.php(174): Piwik\\Plugins\\GoogleAnalyticsImporter\\Importer->importEntities('2', 'YYYYYYY', 'UA-XXXXXXXX-X', 'XXXXXX')\n#9 \/vendor\/symfony\/console\/Symfony\/Component\/Console\/Command\/Command.php(257): Piwik\\Plugins\\GoogleAnalyticsImporter\\Commands\\ImportReports->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#10 \/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(874): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#11 \/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(195): Symfony\\Component\\Console\\Application->doRunCommand(Object(Piwik\\Plugins\\GoogleAnalyticsImporter\\Commands\\ImportReports), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#12 [internal function]: Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#13 \/core\/Console.php(140): call_user_func(Array, Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#14 \/core\/Access.php(644): Piwik\\Console->Piwik\\{closure}()\n#15 \/core\/Console.php(141): Piwik\\Access::doAsSuperUser(Object(Closure))\n#16 \/core\/Console.php(93): Piwik\\Console->doRunImpl(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#17 \/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(126): Piwik\\Console->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#18 \/console(32): Symfony\\Component\\Console\\Application->run()\n#19 {main}"}
```

**Comment by diosmosis** [comment]
I'm not sure if we can detect it, but I can take a look.

**Comment by tsteur** [comment]
@diosmosis we're getting this error a lot every day. Maybe for now when this happens we could at least abort the import? 

Can we show any error why it was aborted?

**Comment by diosmosis** [comment]
@tsteur we can catch it/detect it then show a custom error message, will do that

**Comment by tsteur** [comment]
great, thanks

**Comment by diosmosis** [comment]
Fixed by #122
